### ISSUE TITLE
Recommended way to count number of compilations for a `filter_jit` function?

### ISSUE THREAD

**Issue by paganpasta** [patrick-kidger/equinox#151]
Hi, 

Is there a way to keep a track of different number of times a decorated function (`@eqx.filter_jit`) was recompiled?

The solution [here](https://stackoverflow.com/questions/70126574/jax-hook-information-warning-when-a-jit-function-is-re-compiled) uses a nonlocal variable counter. Is there a recommended way to do this with `equinox` APIs?
The proposed solution produces ` AttributeError: '_JitWrapper' object has no attribute '_cache_size'`.

**Comment by patrick-kidger** [comment]
Right. So `eqx.filter_jit(f)(x)` internally lowers to something that looks a bit like
```python
def call(f, x):
    return f(x)

call_jit = jax.jit(call, static_argnums=0)

call_jit(f, x)
```
which in particular has only a single cache counter, shared across other wrapped functions.

Now technically there's a few more things going on here so that the jit'd function has the correct `__name__` (for use in error messages), which means that the cache counter is at least scoped to wrapped functions with the same name. Whether or not that corresponds to an individual function isn't something that can/should be relied on though.

I think the correct approach is just to use a nonlocal variable counter, as in that question (rather than the answer). (See for example how `chex.assert_max_traces` works if you really want to dial this up to eleven.)

**Comment by paganpasta** [comment]
Ahh yes, counter as in the question. 
Thanks for the helpful answer as always!
